apps/services/user-profile/src/app/worker/test/worker.spec.ts

```diff
@@ -0,0 +1,569 @@
+
+  it('should import new profiles', async () => {
+    // Arrange
+    const advaniaProfiles = new Array(10).fill(null).map((_, index) => ({
+      ssn: `${index}`,
+      email: `test${index}@test.local`,
+      mobilePhoneNumber: `888111${index}`,
+      exported: new Date(2023, 10, 1),
+    }))
+
+    // Act
+    await userProfileAdvaniaModel.bulkCreate(advaniaProfiles)
+    const advaniaProfilesBeforeMigration =
+      await userProfileAdvaniaModel.findAll()
+    const userProfilesBeforeMigration = await userProfileModel.findAll()
+    await workerService.run()
+    const advaniaProfilesAfterMigration =
+      await userProfileAdvaniaModel.findAll()
+    const userProfilesAfterMigration = await userProfileModel.findAll()
+
+    // Assert
+    expect(
+      advaniaProfilesBeforeMigration.every(
+        (p) => p.status === ProcessedStatus.PENDING,
+      ),
+    ).toBe(true)
+    expect(userProfilesBeforeMigration.length).toBe(0)
+    expect(
+      advaniaProfilesAfterMigration.every(
+        (p) => p.status === ProcessedStatus.DONE,
+      ),
+    ).toBe(true)
+    expect(
+      userProfilesAfterMigration.every((p) => {
+        const matchingProfile = advaniaProfilesAfterMigration.find(
+          (ap) => ap.ssn === p.nationalId,
+        )
+
+        return (
+          matchingProfile.email === p.email &&
+          matchingProfile.mobilePhoneNumber === p.mobilePhoneNumber
+        )
+      }),
+    ).toBe(true)
+  })
+
+  describe('profiles being migrated exist in user_profile', () => {
+    it('should keep profiles as is if email and phone numbers match but set nudge to nudgeLastAsked', async () => {
+      // Arrange
+      const advaniaProfiles = new Array(10).fill(null).map((_, index) => ({
+        ssn: `${index}`,
+        email: `test${index}@test.local`,
+        mobilePhoneNumber: `888111${index}`,
+        exported: new Date(2023, 10, 1),
+        nudgeLastAsked: new Date(2023, 10, 5),
+      }))
+
+      const existingProfiles = advaniaProfiles.map((p) => ({
+        nationalId: p.ssn,
+        email: p.email,
+        mobilePhoneNumber: p.mobilePhoneNumber,
+        modified: new Date(2023, 0, 1),
+        lastNudge: new Date(), // not null
+      }))
+
+      // Act
+      await userProfileAdvaniaModel.bulkCreate(advaniaProfiles)
+      await userProfileModel.bulkCreate(existingProfiles)
+
+      const userProfilesBeforeMigration = await userProfileModel.findAll()
+
+      await workerService.run()
+
+      const userProfilesAfterMigration = await userProfileModel.findAll()
+
+      // Assert
+      const fieldsToCompare = ['nationalId', 'email', 'mobilePhoneNumber']
+
+      expect(userProfilesBeforeMigration.length).toBe(
+        userProfilesAfterMigration.length,
+      )
+
+      for (let i = 0; i < userProfilesBeforeMigration.length; i += 1) {
+        const profileBefore = userProfilesBeforeMigration[i]
+        const profileAfter = userProfilesAfterMigration[i]
+        const advaniaProfile = advaniaProfiles[i]
```

Comment:
This depends on the findAll method to return the same order, wondering if that is guaranteed or if this could be flaky and we should set explicit order in findAll? And it needs to be in the same order as the `advaniaProfiles` array.